internal/internal_task_handlers.go

```diff
@@ -402,7 +403,16 @@ func newWorkflowTaskHandler(
 		contextPropagators:             params.ContextPropagators,
 		tracer:                         params.Tracer,
 		workflowInterceptorFactories:   params.WorkflowInterceptorChainFactories,
+		enableStrictNonDeterminism:     params.WorkerBugPorts.EnableStrictNonDeterminismCheck,
 	}
+
+	traceLog(func() {
```

Comment:
I always forget that this exists...

I think it's fine here either way so I'm not requesting any change, but to think out loud I do sorta think this whole concept is better built as either:
- a const, so it can be compile-time elided.  you can change these with build tags.
- just log, and rely on the logger to do the level check efficiently (it does).  should be particularly fine with a non-sugared logger like here.

Reply:
There's a global `var enableVerboseLogging = false` which determines whether this gets logged. there's no level lower than DEBUG in zap so this has been the workaround original authors went with. I used other workaround in the past which required having own wrapper around zap and expose `Trace()` functions and filtering was based on caller file name controlled by env var. 
If you have other ideas we should discuss as part of v2 redesign/refactoring. Leaving this for now as it was still useful to have this option during local testing.